x-pack/plugins/reporting/server/routes/diagnostic/config.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import numeral from '@elastic/numeral';
+import { defaults, get } from 'lodash';
+import { ReportingCore } from '../..';
+import { API_DIAGNOSE_URL } from '../../../common/constants';
+import { LevelLogger as Logger } from '../../lib';
+import { DiagnosticResponse } from '../../types';
+import { authorizedUserPreRoutingFactory } from '../lib/authorized_user_pre_routing';
+
+const KIBANA_MAX_SIZE_BYTES_PATH = 'csv.maxSizeBytes';
+const ES_MAX_SIZE_BYTES_PATH = 'http.max_content_length';
+
+export const registerDiagnoseConfig = (reporting: ReportingCore, logger: Logger) => {
+  const setupDeps = reporting.getPluginSetupDeps();
+  const userHandler = authorizedUserPreRoutingFactory(reporting);
+  const { router, elasticsearch } = setupDeps;
+
+  router.post(
+    {
+      path: `${API_DIAGNOSE_URL}/config`,
+      validate: {},
+    },
+    userHandler(async (user, context, req, res) => {
+      const warnings = [];
+      const { callAsInternalUser } = elasticsearch.legacy.client;
+      const config = reporting.getConfig();
+
+      const elasticClusterSettingsResponse = await callAsInternalUser('cluster.getSettings', {
+        includeDefaults: true,
+      });
+      const { persistent, transient, defaults: defaultSettings } = elasticClusterSettingsResponse;
+      const elasticClusterSettings = defaults({}, persistent, transient, defaultSettings);
+
+      const elasticSearchMaxContent = get(
+        elasticClusterSettings,
+        'http.max_content_length',
+        '100mb'
+      );
+      const elasticSearchMaxContentBytes = numeral().unformat(
+        elasticSearchMaxContent.toUpperCase()
+      );
```

Comment:
TODO for later maybe: use `ByteSizeValue.parse()`, which would use the imported class from `'@kbn/config-schema'` instead of `numeral`.

That might be good: I think this is the only place in Reporting that uses numeral.

Reply:
added the todo to: https://github.com/elastic/kibana/pull/74202/files#diff-ecbc816b7b52fa365baddbe9a6deb730R31